### ISSUE TITLE
New Shortcuts won't check if a dialog is visible

### ISSUE THREAD

**Issue by neikeq** [godotengine/godot#5174]
**Issue description** (what happened, and what was expected):

Check issue: https://github.com/godotengine/godot/pull/3356

The code I used there could be applied to `BaseButton` like this:

``` diff
@@ -405,7 +405,7 @@ Ref<ShortCut> BaseButton:: get_shortcut() const {
 
 void BaseButton::_unhandled_input(InputEvent p_event) {
 
-	if (!is_disabled() && is_visible() && shortcut.is_valid() && shortcut->is_shortcut(p_event)) {
+	if (!is_disabled() && is_visible() && shortcut.is_valid() && shortcut->is_shortcut(p_event) && !get_viewport()->gui_has_modal_stack()) {
 		if (is_toggle_mode()) {
 			set_pressed(!is_pressed());
 			emit_signal("toggled",is_pressed());
```

However, this will remove the possibility to place buttons with shortcuts inside a dialog. There have to be a better way to do it.

**Comment by reduz** [comment]
I don't understand the specific problem, it should be ok that shortcuts don't work if a dialog is visible.. or do you actually mean that they still happen despite the dialog being visible?

**Comment by neikeq** [comment]
@reduz Yes, that's what I meant. They still happen despite another dialog being visible. I fixed this in the past with `!get_viewport()->gui_has_modal_stack()`, e.g: like F5 for running an scene and refreshing a FileDialog. But now it happens again with ShortCut.
However, this time `!get_viewport()->gui_has_modal_stack()` is not a valid solution, since ShortCuts can be applied to buttons in a dialog too.

**Comment by neikeq** [comment]
I have been thinking and this may be the right solution:

``` diff
@@ -405,7 +405,7 @@ Ref<ShortCut> BaseButton:: get_shortcut() const {
 
 void BaseButton::_unhandled_input(InputEvent p_event) {
 
-	if (!is_disabled() && is_visible() && shortcut.is_valid() && shortcut->is_shortcut(p_event)) {
+	if (!is_disabled() && is_visible() && shortcut.is_valid() && shortcut->is_shortcut(p_event) && is_window_modal_on_top()) {
 		if (is_toggle_mode()) {
 			set_pressed(!is_pressed());
 			emit_signal("toggled",is_pressed());
```

`is_window_modal_on_top()` returns true if the current control is the the dialog on top, otherwise false. It also returns false if there is no dialog (`!gui_has_modal_stack()`).

WDYT?

**Comment by akien-mga** [comment]
I guess this is also fixed by ba5bc578?

**Comment by neikeq** [comment]
This issue was fixed by ba5bc57
